src/test/scala/apps/cameraPipeCheck.scala

```diff
@@ -105,12 +112,53 @@ float clamp_f32(float v, float l, float h) {
 #define pow_f32 powf
 """
 
+  val insertCopyAtEnd: Strategy[Rise] =
+  // if...
+    `try`(
+      // ...there is no computation in the AST...
+      predicate.not(oncetd(isComputation)) `;`
+        // ...insert copy: .
+        oncetd(
+          // ...skip all (Dep-)Lambdas...
+          // ( <+ can be read as `or` ) -> maybe introduce `||` notation?
+          predicate.not(isDepLambda <+ isLambda) `;`
+            // ...insert Id() and transform it to an actual copy
+            idAfter `;` normalize.apply(liftId) `;` oncetd(idToCopy)
+        )
+    ) // end-try
+
+  def allIf(cond: Strategy[Rise], s: Strategy[Rise]): Strategy[Rise] =
+    (cond `;` strategies.traversal.all(s)) <+ basic.id[Rise]()
+
+  val p = predicate.not(isApplied(isApplied(isMakeArray)))
+
+  case class traverseWhile(predicate: Strategy[Rise],
+                           s: Strategy[Rise]) extends Strategy[Rise] {
+    def apply(p: Rise): RewriteResult[Rise] =
+      (`try`(s) `;`
+       `if`(predicate)
+          (strategies.traversal.all(traverseWhile(predicate, s))))(p)
+  }
+
+  // todo this is not doing what I want yet!
+  // it should stop traversing as soon as it hits an Apply(makeArray, arg)
```

Comment:
I don't think that looking for `makeArray` should be your goal. I think you should keep track of array dimensions that should be unrolled somehow :thinking: 
Maybe `makeArray` should also be able to not be unrolled .. but then it would probably be sugar for a construct using `generate` and `select`s instead of a separate primitive.